torchvision/models/vision_transformer.py

```diff
@@ -284,7 +285,15 @@ def _vision_transformer(
     progress: bool,
     **kwargs: Any,
 ) -> VisionTransformer:
-    image_size = kwargs.pop("image_size", 224)
+
+    image_size = None
+    if "image_size" in kwargs:
+        image_size = kwargs.pop("image_size", None)
+    if image_size is None and weights is not None and "size" in weights.meta:
+        image_size = weights.meta["size"]
+        if isinstance(image_size, abc.Sequence):
+            image_size = image_size[0]
+    image_size = image_size or 224
```

Comment:
The weights for SWAG vit_b_16 tied to `image_size=384`. I edited this part so the users can just call:
```
m = vit_b_16(ViT_B_16_Weights.IMAGENET1K_SWAG_V1)
```
instead of
```
m = vit_b_16(ViT_B_16_Weights.IMAGENET1K_SWAG_V1, image_size=384)
```
However, there might be a better way to handle this. Any suggestions are welcome :)

Reply:
That's a good approach. I would only refactor slightly to simplify (use if/elif/else). Other than that, your approach looks good.